src/Datadog.Trace/Logging/CustomSerilogLogProvider.cs

```diff
@@ -5,44 +5,83 @@
 
 using System;
 using System.Linq.Expressions;
+using System.Reflection;
+using System.Runtime.CompilerServices;
 using Datadog.Trace.Logging.LogProviders;
 
 namespace Datadog.Trace.Logging
 {
     internal class CustomSerilogLogProvider : SerilogLogProvider, ILogProviderWithEnricher
     {
         private static Func<object, IDisposable> _pushMethod;
+        private static NoOpDisposable _cachedDisposable;
```

Comment:
You don't actually need to store it since it will be kept alive by the lambda closure

Reply:
So you're saying I could use a local and then reference it in my lambda? I just want to make sure the `_pushMethod` is returning the same object over and over, not creating a new one

Reply:
I did that in fbc3aa2030f909177e874b4cb813f518fe687901 , let me know with a 👍🏼 if that looks fine to you. We shouldn't ever run into this scenario because of the new `IsLoggerAvailable()` check, but it's there just in case

Reply:
Yep, that's fine :)